bot/exts/evergreen/fun.py

```diff
@@ -87,6 +92,90 @@ def conversion_func(text: str) -> str:
             converted_text = f">>> {converted_text.lstrip('> ')}"
         await ctx.send(content=converted_text, embed=embed)
 
+    @commands.group(name="caesarcipher", aliases=("caesar", "cc",))
+    async def caesarcipher_group(self, ctx: Context) -> None:
+        """
+        Translates a message using the Caesar Cipher.
+
+        See `decrpyt`, `encrypt`, and `info` subcommands.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.invoke(self.bot.get_command("help"), "caesarcipher")
+
+    @caesarcipher_group.command(name="info")
+    async def caesarcipher_info(self, ctx: Context) -> None:
+        """Information about the Caesar Cipher."""
+        embed = Embed.from_dict(self._caesar_cipher_embed)
+        embed.colour = Colours.dark_green
+
+        await ctx.send(embed=embed)
+
+    @staticmethod
+    async def _caesar_cipher(ctx: Context, offset: int, msg: Union[Message, str]) -> None:
+        """
+        Given an integer `offset`, translates and sends the given `text`.
+
+        A positive `offset` will cause the letters to shift right, while
+        a negative `offset` will cause the letters to shift left.
+
+        Also accepts a valid Discord Message ID or link.
+        """
+        def caesar_func(text: str) -> Iterable[str]:
+            """Implements a lazy Caesar Cipher algorithm."""
+            for char in text:
+                if not char.isascii() or not char.isalpha() or char.isspace():
+                    yield char
+                    continue
+                case_start = 65 if char.isupper() else 97
+                yield chr((ord(char) - case_start + offset) % 26 + case_start)
+
+        def conversion_func(text: str) -> str:
+            """Encrypts the given string using the Caesar Cipher."""
+            return "".join(caesar_func(text))
+
+        is_message = isinstance(msg, Message)
+
+        text = msg.content if is_message else msg
+        embed = msg.embeds[0] if is_message and msg.embeds else None
```

Comment:
You can use `Fun._get_text_and_embed` instead here, similar to how `Fun.uwu_command` does it.

Reply:
I figured that it would be a bit redundant to use said method to manually convert the `msg` that's already handled by the `Union[Message, str]` converter. I

Reply:
Hmm.. I see. It seems pretty redundant doing it either way then. Personally, I think I'd stick with using `Fun._get_text_and_embed` and just have the type annotation as `str`, so it's consistent with the other methods.

It's up to you though, you can do what you see fit.